### PR TITLE
Profile Integration and dedicated tool to draw profile's line

### PR DESCRIPTION
**Pull request by ger-benjamin**
Fix: camptocamp/c2cgeoportal#1661
Specs: https://github.com/camptocamp/c2cgeoportal/wiki/Spec-%231661-Profiles

Example: 
 - 1 elevation raster: https://ger-benjamin.github.io/ngeo/profile21/examples/contribs/gmf/apps/desktop_alt
 - Multiple elevations rasters: https://ger-benjamin.github.io/ngeo/profile21/examples/contribs/gmf/apps/desktop

(See just the last commit for the intergration and the new tool)

@pgiraud I need some specifications for the "Draw profile tool", the activation, the placement and some differences that come from the profile with more than 1 line. Can you take a look please ?
Edit: Integrated. Thanks

---

**Issue to create and resolve later** :
- Reload the profile on screen size change.
- It exists any tests for ngeo and gmf profile.

### CODE REVIEW

**Comment by ger-benjamin** [comment]
Please review

**Comment by ger-benjamin** [comment]
Ping :cake:

**Comment by fredj** [comment]
reviewing now

**Comment by fredj** [comment]
When the profile panel is active, the query tool must be deactivated (see #1478)

File: contribs/gmf/src/directives/profile.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ ngeo.module.value('gmfProfileTemplateUrl',
  *      </gmf-profile>
  *
  *
- * @htmlAttribute {boolean?} gmf-profile-active Optional to active the hover
+ * @htmlAttribute {boolean} gmf-profile-active To active th
```
Comment (fredj) [inline contribs/gmf/src/directives/profile.js]: `the` instead of `th`

**Comment by fredj** [comment]
LGTM, it would be nice that the d3 graph is resized with the page but this can be done later

**Comment by ger-benjamin** [comment]
@fredj Can you take a look on my last commit ?

File: contribs/gmf/src/controllers/abstractdesktop.js [hunk 2]
```diff
@@ -142,6 +144,21 @@ gmf.AbstractDesktopController = function(config, $scope, $injector) {
...
+   * @type {boolean}
+   * @export
+   */
+  this.drawProfilePanelActive = false;
+
+  /**
+   * The ngeo ToolActivate manager service.
+   * @type {ngeo.ToolActivateMgr}
+   */
+  var ngeoToolActivateMgr = $injector.get('ngeoToolActivateMgr');
```
Comment (fredj) [inline contribs/gmf/src/controllers/abstractdesktop.js]: Even if the tool is not present in the mobile app, all this code should be moved to `contribs/gmf/src/controllers/abstract.js`

**Comment by ger-benjamin** [comment]
Done. Thanks @fredj @pgiraud can I merge ?

**Comment by fredj** [comment]
ok for me

**Pull request merged**